homeassistant/components/aosmith/__init__.py

```diff
@@ -11,33 +9,37 @@
 from homeassistant.helpers import aiohttp_client
 
 from .const import DOMAIN
-from .coordinator import AOSmithCoordinator
+from .coordinator import AOSmithEnergyCoordinator, AOSmithStatusCoordinator
+from .models import AOSmithData, build_device_details
 
 PLATFORMS: list[Platform] = [Platform.SENSOR, Platform.WATER_HEATER]
 
 
-@dataclass
-class AOSmithData:
-    """Data for the A. O. Smith integration."""
-
-    coordinator: AOSmithCoordinator
-    client: AOSmithAPIClient
-
-
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up A. O. Smith from a config entry."""
     email = entry.data[CONF_EMAIL]
     password = entry.data[CONF_PASSWORD]
 
     session = aiohttp_client.async_get_clientsession(hass)
     client = AOSmithAPIClient(email, password, session)
-    coordinator = AOSmithCoordinator(hass, client)
 
-    # Fetch initial data so we have data when entities subscribe
-    await coordinator.async_config_entry_first_refresh()
+    status_coordinator = AOSmithStatusCoordinator(hass, client)
+    await status_coordinator.async_config_entry_first_refresh()
+
+    energy_coordinator = AOSmithEnergyCoordinator(
+        hass, client, list(status_coordinator.data.keys())
```

Comment:
```suggestion
        hass, client, list(status_coordinator.data)
```
I _think_ this is the same